lib/kafka/protocol/message_set.rb

```diff
@@ -3,9 +3,14 @@ module Protocol
     class MessageSet
       attr_reader :messages
 
-      def initialize(messages: [], compression_codec: nil)
+      def initialize(messages: [], compression_codec: nil, compression_threshold: 1)
```

Comment:
I’m looking at that `nil` and wondering we should pass a null codec as the default, and reduce `#compress?` to just compare the size vs threshold.?